pkg/datagatherer/k8s/dynamic_test.go

```diff
@@ -582,19 +598,55 @@ func TestDynamicGatherer_Fetch(t *testing.T) {
 				},
 			},
 		},
+		"excluded annotations are removed for unstructured-based gatherers such as secrets": {
+			config: ConfigDynamic{GroupVersionResource: schema.GroupVersionResource{Group: "", Version: "v1", Resource: "secrets"}},
+
+			// To give a realistic regex in this test case, let's use the
+			// example of the Kapp project that uses four annotations that all
+			// start with `kapp.k14s.io/original*`. These annotations are
+			// similar to `kubectl.kubernetes.io/last-applied-configuration` in
+			// that they may contain sensitive information. From [1], they may
+			// look like this:
+			//
+			//  kapp.k14s.io/original: |
+			//    {"apiVersion":"v1","kind":"Secret","spec":{"data": {"password": "cGFzc3dvcmQ=","username": "bXl1c2VybmFtZQ=="}}}
+			//  kapp.k14s.io/original-diff: |
+			//    - type: test
+			//      path: /data
+			//      value:
+			//      password: cygpcGVyUzNjcmV0UEBhc3N3b3JkIQ==
+			//      username: bXl1c2VybmFtZQ==
+			//
+			//  [1]: https://github.com/carvel-dev/kapp/issues/90#issuecomment-602074356
+			excludeAnnotsKeys: []string{`kapp\.k14s\.io\/original.*`},
```

Comment:
I don't know why I escaped the character `/`. I'll un-escape it.

```suggestion
			excludeAnnotsKeys: []string{`kapp\.k14s\.io/original.*`},
```

Reply:
I also forgot to use `^`. I'll add it too.

```suggestion
			excludeAnnotsKeys: []string{`^kapp\.k14s\.io\/original.*`},
```

Reply:
Done.